Sources/Contour/Transport/RabbitMQ/Internal/Listener.cs

```diff
@@ -434,6 +432,11 @@ private void ConsumerTaskMethod(CancellationToken token)
             {
                 var consumer = this.InitializeConsumer(token, out var channel);
 
+                // если шина так и не стала готова работать, то не смысла начинать слушать сообщения, что бы потом их потерять
+                this.busContext.WhenReady.WaitOne();
```

Comment:
Возможно, имеет смысл вписать в WaitOne какой-то ограниченный таймаут? Например, 60 секунд.
```suggestion
                if (!this.busContext.WhenReady.WaitOne(60000)) then throw;
```

Reply:
в таком случае. если при инициализации шины произойдет ошибка, то через 60 секунд начнем терять сообщения с ошибкой бас нот реди, как происходит сейчас, вместо их накопления

Reply:
так же это сломает механизм переподключения, если нам 60 секунд не хватило для этого

Reply:
Ладно. Как насчет варианта писать WARN в лог при ожидании больше 60 секунд?

Reply:
А где переводится примитив в сигнальное состояние? Сейчас при создании консьюмера зайдет поток, захватит блокировку и переведет в несигнальное. Пока в сигнальное состояние не будет переведен примитив, то новый поток повиснет.

Reply:
Хорошая мысль, как это правильно сделать? сначала подождать 60 сек, записать варн и ждать до упора потом?

Reply:
> А где переводится примитив в сигнальное состояние? Сейчас при создании консьюмера зайдет поток, захватит блокировку и переведет в несигнальное. Пока в сигнальное состояние не будет переведен примитив, то новый поток повиснет.
 вот тут
https://github.com/SDVentures/Contour/blob/core/Sources/Contour/Transport/RabbitMQ/Internal/RabbitBus.cs#L220

Флаг будет взведен после того, как все листнеры и паблишеры будут инициализированны

Reply:
> Ладно. Как насчет варианта писать WARN в лог при ожидании больше 60 секунд?

Хорошая мысль, как это правильно сделать? сначала подождать 60 сек, записать варн и ждать до упора потом?

Reply:
Я бы сделал отдельный поток, который так же ждет события, но 60 секунд, и пишет warn, если не дождался.

Reply:
Хотя наверное это лишнее. Даже если между двумя вызовами WaitOne и произойдет set + reset, это все равно будет означать, что нам дальше нужно ждать. Или я не прав?

Reply:
> Хотя наверное это лишнее. Даже если между двумя вызовами WaitOne и произойдет set + reset, это все равно будет означать, что нам дальше нужно ждать. Или я не прав?

да, всеравно надо продолжить ждать. И такой сценарий маловероятен

Reply:
> А где переводится примитив в сигнальное состояние? Сейчас при создании консьюмера зайдет поток, захватит блокировку и переведет в несигнальное. Пока в сигнальное состояние не будет переведен примитив, то новый поток повиснет.

Так же добавлю, что все задачи создаются с опцией LongRunning, и если я верно понимаю, что написано в документации, то такая задача не должна блокировать другие потоки:
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcreationoptions?view=netstandard-2.0